src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```
@@ -2,21 +2,47 @@
 
 import java.util.Optional;
 import mate.academy.hibernate.relations.dao.MovieDao;
+import mate.academy.hibernate.relations.exception.DataProcessingException;
 import mate.academy.hibernate.relations.model.Movie;
-import org.hibernate.SessionFactory;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
 
 public class MovieDaoImpl extends AbstractDao implements MovieDao {
-    public MovieDaoImpl(SessionFactory sessionFactory) {
+    public MovieDaoImpl(org.hibernate.SessionFactory sessionFactory) {
         super(sessionFactory);
     }
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Transaction tx = null;
+        try (Session session = factory.openSession()) {
+            tx = session.beginTransaction();
+            session.save(movie);
+            tx.commit();
+            return movie;
+        } catch (Exception e) {
+            if (tx != null) {
+                try {
+                    tx.rollback();
+                } catch (Exception ignored) {
```

Ignoring the exception during a transaction rollback can hide critical database issues. It's better to handle this exception, for instance by logging it or wrapping it in a `DataProcessingException` as you've done in `ActorDaoImpl`, to ensure that the calling code is aware of all potential failures.